bluemira/builders/EUDEMO/tools.py

```diff
@@ -280,6 +281,26 @@ def _order_shapes_anticlockwise(shapes):
     return list(np.array(shapes)[indices])
 
 
+def find_xy_plane_radii(wire, plane):
+    """
+    Get the radial coordinates of a wire's intersection points with a plane.
+
+    Parameters
+    ----------
+    wire: BluemiraWire
+        Wire to get the radii for in the plane
+    plane: BluemiraPlacement
+        Plane to slice with
+
+    Returns
+    -------
+    radii: list
+        The array of radii of intersections, sorted from smallest to largest
+    """
+    intersections = slice_shape(wire, plane)
+    return sorted(intersections[:, 0])
```

Comment:
I'm still a little concerned with the usage of `intersections` here and the return type of `slice_shape`. Are we guaranteed not to get two return values from `slice_shape` because we're slicing a wire? Also, are we guaranteed that `intersections` won't be `None`?

I think the API of `slice_shape` is a little tricky, really. It's putting a lot of burden on the caller to check all the different possible return types, without saying in the docstring what outputs I can expect from what inputs. I'm not even sure how many outputs I should expect.

This is a job for a separate issue, but I really think splitting `slice_shape` into separate functions `slice_wire`, `slice_face`, etc. with clear return types would reduce the load on the caller.

Reply:
https://github.com/Fusion-Power-Plant-Framework/bluemira/issues/1050